src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -6,227 +6,199 @@
- * types have a look at [[StringConvert]]
+ * Trait for conversion between `T` and `ConfigValue`.
  */
 trait ConfigConvert[T] {
   /**
-   * Convert the given configuration into an instance of `T` if possible
+   * Convert the given configuration into an instance of `T` if possible.
    *
    * @param config The configuration from which load the config
-   * @param namespace The base namespace to use for the conversion. This should be used as base
-   *                  for the name of the fields of `T`. For instance, given a `case class Foo(i: Int)`
-   *                  and a configuration `conf = Map("foo.i", "1")`, then `from(conf, "foo")` will return
-   *                  `Success(Foo(1))` while `from(conf, "")` will return `Failure`
    * @return `Success` of `T` if the conversion is possible, `Failure` with the problem if the
    *         conversion is not
    */
-  def from(config: RawConfig, namespace: String): Try[T]
+  def from(config: ConfigValue): Try[T]
 
   /**
-   * Converts a type `T` to a `RawConfig` using the `namespace` as base namespace.
+   * Converts a type `T` to a `ConfigValue`.
    *
    * @param t The instance of `T` to convert
-   * @param namespace The base namespace. For instance, given case `class Foo(i: Int)`, then
-   *                  `to(Foo(2), "base.namespace")` will return `Map("base.namespace.i", "2")`
-   * @return The `RawConfig` obtained from the `T` instance
+   * @return The `ConfigValue` obtained from the `T` instance
    */
-  def to(t: T, namespace: String): RawConfig
+  def to(t: T): ConfigValue
 }
 
 object ConfigConvert extends LowPriorityConfigConvertImplicits {
   def apply[T](implicit conv: ConfigConvert[T]): ConfigConvert[T] = conv
 
+  private def fromFConvert[T](fromF: String => T): ConfigValue => Try[T] =
+    config => {
+      Try(fromF(config.valueType() match {
+        case ConfigValueType.STRING => config.unwrapped().toString
+        case _ => config.render(ConfigRenderOptions.concise())
+      }))
+    }
+
+  def fromString[T](fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Try[T] = fromFConvert(fromF)(config)
+    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
+  }
+
+  def stringConvert[T](fromF: String => T, toF: T => String): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Try[T] = fromFConvert(fromF)(config)
+    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(toF(t))
+  }
+
   implicit def hNilConfigConvert = new ConfigConvert[HNil] {
-    override def from(config: RawConfig, namespace: String): Try[HNil] = Success(HNil)
-    override def to(t: HNil, namespace: String): RawConfig = Map.empty[String, String]
+    override def from(config: ConfigValue): Try[HNil] = Success(HNil)
+    override def to(t: HNil): ConfigValue = ConfigFactory.parseMap(Map().asJava).root()
```

Comment:
I wonder if there is an easier way to get a `ConfigObject` representing an empty map that doesn't require to call `parseMap(..)` and then `root()`. I couldn't find anything.